packages/dev/core/src/Engines/engine.ts

```diff
@@ -421,7 +421,19 @@ export class Engine extends ThinEngine {
     /**
      * Gets a boolean indicating if the pointer is currently locked
      */
-    public isPointerLock = false;
+    public get isPointerLock() {
+        // Return either manual pointerlock state or canvas pointerlock state
+        return this._isPointerLock || (this._isCanvasPresent && document.pointerLockElement === this._renderingCanvas);

```

Comment:
Where is _isPointerLock ever set?

Reply:
Following up on this - I am not sure how this will solve the issue.
isPointerLock set to true will forever return true. It is never set to false? The dev will need to be responsible for setting this to false. Is that the expected behavior? because until now it was simply a flag to be read by the dev.

Reply:
The expected behavior with this is that the user _can_ set `isPointerLock` (effectively setting `_isPointerLock`) but generally speaking they shouldn't.  The reason why I did it this way was because there are specific scenarios like allowing the Babylon Native (on the Native side) to control the pointerlock.  If the user decides to manually set this to `true`, they will have to then change it to `false` manually.

The behavior doesn't necessarily need to be set this way.  This was just some forward planning if we ever get around to adding a native pointerlock behavior.

Reply:
So the dev can "mock" pointerlock behavior if it is set to true, otherwise it will be set automatically?

Reply:
Side note - if this is the suggested behavior, this will need to be well documented. Until now setting the pointerLock would not have any effect, though possible

Reply:
> Until now setting the pointerLock would not have any effect, though possible

I'm not sure I understand what this means. `pointerLock` is already a settable boolean before and it does have an effect. Am I misunderstanding something?

Reply:
isPointerLock is set automatically from false to true and back again, without the need for dev interference. You could set it yourself, but there is no actual reason to do that (on the web. I understand that it is required in native). in the version I was discussing, if it is ever set to true, it will forever stay true, even if the user was no longer in a pointer lock.

Reply:
> on the web. I understand that it is required in native

There should be no additional requirement to make it work in native other than the document/canvas being missing. All other behavior should be the same as the web.